resp/resp3/resp.go

```diff
@@ -1677,27 +1679,29 @@ func Unmarshal(br resp.BufferedReader, rcv interface{}, o *resp.Opts) error {
 	}
 	prefix := Prefix(b[0])
 
-	// if the prefix is one of the error types then just parse and return that
-	// full message here using the actual unmarshalers, which is easier than
-	// re-implementing them.
-	switch prefix {
-	case SimpleErrorPrefix:
-		var into SimpleError
-		if err := into.UnmarshalRESP(br, o); err != nil {
-			return err
-		}
-		return resp.ErrConnUsable{Err: into}
-	case BlobErrorPrefix:
-		var into BlobError
-		if err := into.UnmarshalRESP(br, o); err != nil {
-			return err
-		}
-		return resp.ErrConnUsable{Err: into}
-	case AttributeHeaderPrefix:
-		if err := DiscardAttribute(br, o); err != nil {
-			return err
+	if o == nil || !o.TreatErrorsAsValues {
```

Comment:
Again, opts should never be nil (I don't think? you've looked at this code more recently though lol)

Reply:
Done